BudgetAutomation/BudgetAutomation.Engine/ApiClient/ExpenseLoggerApiClient.cs

```diff
@@ -76,6 +84,40 @@ public async Task<Expense> LogExpenseAsync(
             _logger.LogError("Received status code {StatusCode}, but content was null or not JSON.", response.StatusCode);
         }
 
-        return new Expense();
+        return new LogExpenseResponse();
+    }
+
+    public async Task<SpreadsheetValidatorResponse> ValidateSpreadsheet(SpreadsheetValidatorRequest request)
+    {
+        _logger.LogInformation("Sending request to /validate-spreadsheet");
+
+        var endpointUri = new Uri(_httpClient.BaseAddress!, "validate-spreadsheet");
+
+        var httpResponse = await _httpClient.PostAsJsonAsync(endpointUri, request);
+
+        if (!httpResponse.IsSuccessStatusCode)
+        {
+            _logger.LogError("Validation failed with status code: {StatusCode}", httpResponse.StatusCode);
+            return new SpreadsheetValidatorResponse
+            {
+                Success = false,
+                Message = $"Validation failed with status code: {httpResponse.StatusCode}"
+            };
+        }
+
+        var result = await httpResponse.Content.ReadFromJsonAsync<SpreadsheetValidatorResponse>();
```

Comment:
Deserializa igual o resto do projeto
```suggestion
            var result = await response.Content.ReadFromJsonAsync(AppJsonSerializerContext.Default.SpreadsheetValidatorResponse, cancellationToken);
```